src/shared/services/map-location/index.js

```diff
@@ -1,28 +1,60 @@
-function mapLocation(loc) {
+export const location2feature = location => ({
+  type: 'Point',
+  coordinates: [location.lng, location.lat],
+});
+
+export const feature2location = feature => {
+  const { coordinates } = feature;
+  return {
+    lat: coordinates[1],
+    lng: coordinates[0],
+  };
+};
+
+export const address2pdok = address => {
+  const { openbare_ruimte, huisnummer, huisletter, huisnummer_toevoeging, postcode, woonplaats } = address;
+
+  return {
+    straatnaam: openbare_ruimte,
+    huisnummer: `${huisnummer}`,
+    huisletter: huisletter || '',
+    huisnummertoevoeging: huisnummer_toevoeging ? String(huisnummer_toevoeging) : '',
+    postcode,
+    woonplaatsnaam: woonplaats,
+  };
+};
+
+/**
+ * converts the location from `sia` location format to latlon format
+ */
+export function mapLocation(loc) {
   const location = {};
 
-  if (loc.dichtstbijzijnd_adres) {
-    location.address = { ...loc.dichtstbijzijnd_adres };
-    location.address.huisnummer = `${location.address.huisnummer}`;
-    location.address.huisnummer_toevoeging = `${location.address.huisnummer_toevoeging}`;
+  if (loc.geometrie) {
+    location.location = feature2location(loc.geometrie);
   }
 
-  if (loc.omgevingsinfo) {
-    location.buurt_code = loc.omgevingsinfo.buurtcode;
-    location.stadsdeel = loc.omgevingsinfo.stadsdeelcode;
+  if (loc.buurt_code) {
+    location.buurtcode = loc.buurt_code;
   }
 
-  if (loc.query) {
-    location.geometrie = {
-      type: 'Point',
-      coordinates: [
-        loc.query.longitude,
-        loc.query.latitude,
-      ],
-    };
+  if (loc.stadsdeel) {
+    location.stadsdeelcode = loc.stadsdeel;
+  }
+
+  if (loc.address) {
+    location.address = address2pdok(loc.address);
   }
 
   return location;
 }
 
+export const formatAddress = address => {
```

Comment:
This function doesn't seem to be used. Please remove if not used.

Reply:
it is used in MapPreview